src/jit/treelifeupdater.h

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+
+#include "compiler.h"
+
+//------------------------------------------------------------------------
+// TreeLifeUpdater: class that handles life variables changes according to a tree
+// and updates compiler sets accordingly. Keeps set of temporary VARSET_TP during its lifetime
+// to avoid unnecessary memory allocations.
+template <bool ForCodeGen>
+class TreeLifeUpdater
+{
+public:
+    TreeLifeUpdater(Compiler* compiler);
+    void UpdateLife(GenTree* tree);
+
+private:
+    void UpdateLifeVar(GenTree* tree);
+
+private:
+    Compiler* compiler;
+    VARSET_TP newLife;          // a life set after processing an argument tree.
+    VARSET_TP stackVarDeltaSet; // a life set of tracked stack ptr lcls.
```

Comment:
In the two lines above, I would use "live set".

Reply:
Fixed.